### ISSUE TITLE
Issue with Multiprocessing on MacOS

### ISSUE THREAD

**Issue by yongzx** [bigscience-workshop/promptsource#572]
I run into the issue of `Multiprocessing causes Python to crash and gives an error may have been in progress in another thread when fork() was called`, which I fix (by referring to the [thread](https://stackoverflow.com/questions/50168647/multiprocessing-causes-python-to-crash-and-gives-an-error-may-have-been-in-progr)) by adding `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` to my .zshrc

However, now I encounter another error of `The process has forked and you cannot use this CoreFoundation functionality safely`, of which one of the [solutions](https://stackoverflow.com/questions/30669659/multiproccesing-and-error-the-process-has-forked-and-you-cannot-use-this-corefou) requires modifying the script app.py.

Wondering if there's a quick fix to this issue, or should I modify the app.py and do a PR?

**Comment by stephenbach** [comment]
I started hitting this issue after upgrading OSX to 12.3. When I include the suggested fix, it works in that I can navigate off the helicopter view and use the rest of promptsoruce normally, but the helicopter view itself fails with this error
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/multiprocessing/spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/multiprocessing/spawn.py", line 114, in _main
    prepare(preparation_data)
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/multiprocessing/spawn.py", line 225, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/multiprocessing/spawn.py", line 277, in _fixup_main_from_path
    run_name="__mp_main__")
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/Users/bach/anaconda3/envs/promptsource/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/bach/promptsource/promptsource/app.py", line 67, in <module>
    state = _get_state()
  File "/Users/bach/clones/promptsource/promptsource/session.py", line 84, in _get_state
    session = _get_session()
  File "/Users/bach/clones/promptsource/promptsource/session.py", line 74, in _get_session
    session_id = get_report_ctx().session_id
AttributeError: 'NoneType' object has no attribute 'session_id'
```
(Shows up as EOF error inside streamlit.) Are you seeing this too?

**Comment by stephenbach** [comment]
This seems to happen because when new processes are launched with spawn rather than fork, streamlit gets set up again, which breaks the internal session management. Possibly also the cause of #588 that couldn't be reproduced.

**Comment by stephenbach** [comment]
Sorry for the stream of consciousness updates, but multiprocessing seems to be the source of many of our problems. It's also the reason we can't upgrade to python 3.8 (as described in #416). It seems that that is because 3.8 switches the default launch method to spawn: https://chrissardegna.com/blog/multiprocessing-changes-python-3-8/

**Comment by stephenbach** [comment]
Closed via #733.
